code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/service/impl/ServiceDiscoveryServiceImpl.java

```diff
@@ -321,8 +321,9 @@ private void populateExternalLinksForService(Service service, Map<String, Object
                     ((List<Object>) dataObj).addAll((List<Object>) launchConfigItems.get(key));
                 }
             }
-
-            launchConfigItems.put(key, dataObj);
+            if (dataObj != null) {
+                launchConfigItems.put(key, dataObj);
```

Comment:
service config coming from UI was populated requestedHostId to be null.  This overrode the requestedHostId we figure out from the globalService causing all the containers to go to the same host